src/k8s/pkg/k8sd/pki/control_plane_test.go

```diff
@@ -1,12 +1,20 @@
 package pki_test
 
 import (
+	"crypto/x509"
+	"embed"
+	"encoding/pem"
+	"io/fs"
+	"net"
 	"testing"
 
 	"github.com/canonical/k8s/pkg/k8sd/pki"
 	. "github.com/onsi/gomega"
 )
 
+//go:embed data
+var testCertificates embed.FS
```

Comment:
Test files typically go under `testdata`.

```suggestion
//go:embed testdata
var testdata embed.FS
```

Also consider whether you would rather have something simpler like

```
func mustReadTestData(filename string) string
```

which panics if the file cannot be found.